pkg/instrumentation/redis/redis.go

```diff
@@ -0,0 +1,191 @@
+
+		MaxRedirects:   cfg.MaxRedirects,
+		ReadOnly:       cfg.ReadOnly,
+		RouteByLatency: cfg.RouteByLatency,
+		RouteRandomly:  cfg.RouteRandomly,
+
+		MasterName:       cfg.MasterName,
+		DisableIndentity: cfg.DisableIndentity,
+		IdentitySuffix:   cfg.IdentitySuffix,
+	}
+	if clusterOptions != nil {
+		opts.Addrs = clusterOptions.Addrs
+		opts.ClientName = clusterOptions.ClientName
+
+		opts.Protocol = clusterOptions.Protocol
+		opts.Username = clusterOptions.Username
+		opts.Password = clusterOptions.Password
+
+		if clusterOptions.MaxRetries != 0 {
+			opts.MaxRetries = clusterOptions.MaxRetries
+		}
+		if clusterOptions.MinRetryBackoff != 0 {
+			opts.MinRetryBackoff = clusterOptions.MinRetryBackoff
+		}
+		if clusterOptions.MaxRetryBackoff != 0 {
+			opts.MaxRetryBackoff = clusterOptions.MaxRetryBackoff
+		}
+
+		if clusterOptions.DialTimeout != 0 {
+			opts.DialTimeout = clusterOptions.DialTimeout
+		}
+		if clusterOptions.ReadTimeout != 0 {
+			opts.ReadTimeout = clusterOptions.ReadTimeout
+		}
+		if clusterOptions.WriteTimeout != 0 {
+			opts.WriteTimeout = clusterOptions.WriteTimeout
+		}
+		if clusterOptions.ContextTimeoutEnabled {
+			opts.ContextTimeoutEnabled = clusterOptions.ContextTimeoutEnabled
+		}
+
+		if clusterOptions.PoolSize != 0 {
+			opts.PoolSize = clusterOptions.PoolSize
+		}
+		if clusterOptions.PoolTimeout != 0 {
+			opts.PoolTimeout = clusterOptions.PoolTimeout
+		}
+		if clusterOptions.MaxIdleConns != 0 {
+			opts.MaxIdleConns = clusterOptions.MaxIdleConns
+		}
+		if clusterOptions.MinIdleConns != 0 {
+			opts.MinIdleConns = clusterOptions.MinIdleConns
+		}
+		if clusterOptions.MaxActiveConns != 0 {
+			opts.MaxActiveConns = clusterOptions.MaxActiveConns
+		}
+		if clusterOptions.ConnMaxIdleTime != 0 {
+			opts.ConnMaxIdleTime = clusterOptions.ConnMaxIdleTime
+		}
+		if clusterOptions.ConnMaxLifetime != 0 {
+			opts.ConnMaxLifetime = clusterOptions.ConnMaxLifetime
+		}
+
+		if clusterOptions.MaxRedirects != 0 {
+			opts.MaxRedirects = clusterOptions.MaxRedirects
+		}
+		if clusterOptions.ReadOnly {
+			opts.ReadOnly = clusterOptions.ReadOnly
+		}
+		if clusterOptions.RouteByLatency {
+			opts.RouteByLatency = clusterOptions.RouteByLatency
+		}
+		if clusterOptions.RouteRandomly {
+			opts.RouteRandomly = clusterOptions.RouteRandomly
+		}
+	}
+
+	if cfg.TLS.Enabled {
+		var caCertPool *x509.CertPool
+
+		if cfg.TLS.Ca != "" {
+			caCertPool = x509.NewCertPool()
+			caCertPool.AppendCertsFromPEM([]byte(cfg.TLS.Ca))
+		}
+
+		var certificates []tls.Certificate
+		if cfg.TLS.Cert != "" && cfg.TLS.CertKey != "" {
+			cert, err := tls.X509KeyPair([]byte(cfg.TLS.Cert), []byte(cfg.TLS.CertKey))
+			if err != nil {
+				return nil, err
+			}
+			certificates = []tls.Certificate{cert}
+		}
+
+		if cfg.TLS.Cert != "" && cfg.TLS.CertKey != "" {
+			cert, err := tls.X509KeyPair([]byte(cfg.TLS.Cert), []byte(cfg.TLS.CertKey))
+			if err != nil {
+				return nil, err
+			}
+			certificates = []tls.Certificate{cert}
+		}
```

Comment:
@Neurostep why are we doing this twice? 🤔 

I have a feeling that I'm missing something obvious. 😅

Reply:
@fotos 

>why are we doing this twice? 🤔

Just to be sure 😅  On a serious note, it is a mistake, removed it ✂️